protocol/0030-ETHM-multisig_control_spec.md

```diff
@@ -8,45 +8,60 @@ Vega controls and maintains a number of Ethereum smart contracts which have func
-NOTE: when msg.sender is required by function, that account must also be the submitter of the transaction to the Ethereum blockchain 
-
-Pseudo Example:  
-A bridge smart contract has a `withdraw_asset(address asset_source, uint256 asset_id, uint256 amount, uint256 nonce, bytes memory signatures)` function,
-the function requires that msg.sender be passed along as well (given that it's the wallet address that will be credited with this withdrawal)
-This means that the message to sign will look like this: 
-    `abi.encode(abi.encode(asset_source, asset_id, amount, address_of_user, nonce, "withdraw_asset"), bridge_address)`
-The resulting message is then hashed to a bytes32 message_hash which is then signed by a Vega validator (that has been previously added as a valid signer to the `MultisigControl` smart contract)
- 
-Each signature is a 65 byte hex string in the format `0x1230477...` the `0x` is removed (except for the first in the bundle) and appended to the end of the bundle of other validator signatures. 
-This bundle is the `signatures` parameter ultimately passed to `verify_signatures`. In the `verify_signatures` function this signature bundle is broken into individual signatures and checked against the valid signer list. If the number of valid signatures if above the threshold %, this is a valid bundle.   
-
-## Adding Signers
-Additional Vega validator nodes are added as MultisigControl signers by running the `add_signer` function.
-
-Following the information above from "Signature Bundles", the message to sign will look like this:
-`abi.encode(abi.encode(new_signer_address, nonce, "add_signer"), submitter)`
-In this case, the submitter will most likely be the new signer since they are effectively the proposer. The reason the submitter is not a contract as in the case with the bridges is that the function is called directly by a user, not by a smart contract, like the bridge. 
-
-## Removing Signer
-Vega validator nodes are removed from being MultisigControl signers by running the `remove_signer` function.
-Following the information above from "Signature Bundles", the message to sign will look like this:
-`abi.encode(abi.encode(old_signer_address, nonce, "remove_signer"), submitter)`
-In this case, the submitter will likely be the user/party who proposed the removal of that node as a signer.  
+In order for a smart contract to be securely controlled by the Vega network, it must reference a previously launched and configured `Multisig_Control` smart contract and call `verify_signatures` before executing a Vega controlled function.    
 
+A signature bundle is a hex string of appended 65 byte ECDSA signatures where each signer signed the same message hash usually containing the parameters of the function being called, the current epoch data consisting of a list of signers and their corresponding weights, and a nonce that is used to prevent replay attacks.
+
+## Epoch Data
```

Comment:
I don't think we want to insist on per-epoch updates. 

The validators will have protocol-level incentives to update the multisig weights *if* they weights or the signer set changes.

Reply:
I'll change the wording of it and call it something else